enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/pipes/RollUpApplySlottedPipe.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility.v3_4.runtime.slotted.pipes
+
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime._
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.commands.expressions.Expression
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.pipes.{Pipe, PipeWithSource, QueryState}
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.slotted.PrimitiveExecutionContext
+import org.neo4j.cypher.internal.v3_4.logical.plans.LogicalPlanId
+import org.neo4j.values.AnyValue
+import org.neo4j.values.storable.Values.NO_VALUE
+import org.neo4j.values.virtual.VirtualValues
+
+case class RollUpApplySlottedPipe(lhs: Pipe, rhs: Pipe,
+                                  collectionName: String,
+                                  identifierToCollect: (String, Expression),
+                                  nullableIdentifiers: Set[String],
+                                  pipelineInformation: PipelineInformation)
+                                 (val id: LogicalPlanId = LogicalPlanId.DEFAULT)
+  extends PipeWithSource(lhs) {
+
+  private val collectionSlot = pipelineInformation.get(collectionName).get
+
+  private val getValueToCollectFunction = {
+    val expression: Expression = identifierToCollect._2
+    (state: QueryState) => (ctx: ExecutionContext) => expression(ctx, state)
+  }
+
+  private val hasNullValuePredicates: Seq[(ExecutionContext) => Boolean] =
```

Comment:
I think I would prefer this to be done outside this file and have this file focus on the runtime aspects. When reading this code I want to be in a runtime frame of mind, and here we are mixing between the two.

Pretty sure this is highly subjective, so feel free to ignore me. This is just my humble opinion.

Reply:
I don't really have a strong opinion, but I chose to humbly ignore yours :) Currently there is a mix of two styles across slotted pipe implementations, so maybe we could agree on one style, document it and unify the code as a separate concern.